doc/source/dev/reviewer_guidelines.rst

```diff
@@ -7,19 +7,48 @@ Reviewer Guidelines
 Reviewing open pull requests (PRs) is very welcome and a valuable way to help
 increase the speed at which the project moves forward. We encourage everyone to
 get involved in the review process; it's also a great way to get familiar with
-the codebase. If we do not know you yet, consider introducing yourself before
-you start reviewing code.
+the codebase.
+
+Who can be a reviewer?
+======================
+
+You don't have to be a project member to contribute reviews. Consider being a
+reviewer if 
+
+- You have contributed to the NumPy project before, and feel like you can
+  comment on technical or organizational aspects of the PR; 
+- You haven't contributed before, but are an experienced NumPy user, and see a
+  pull request that touches a funcionality that you use often;
+- You can contribute comments about content, style, organization or good
+  documentation practices.  
+
+If we do not know you yet, consider introducing yourself before you start
+reviewing pull requests.
+
+Communication Guidelines
+========================
+
+- Every PR, good or bad, is an act of generosity. Opening with a positive
+  comment will help the author feel rewarded, and your subsequent remarks may be
+  heard more clearly. You may feel good also.
+- Begin if possible with the large issues, so the author knows they've been
+  understood. Resist the temptation to immediately go line by line, or to open
+  with small pervasive issues.
+- You are the face of the project, and NumPy some time ago decided `the kind of
+  project it will be <https://numpy.org/code-of-conduct/>`_: open, empathetic,
+  welcoming, friendly and patient. Be `kind
+  <https://youtu.be/tzFWz5fiVKU?t=49m30s>`_ to contributors.
+- Do not let perfect be the enemy of the good, particularly for documentation.
+  If you find yourself making many small suggestions, either open a PR against
+  the original branch, push changes to the contributor branch, or merge the PR
+  and then open a new PR against upstream.
+- If you need help writing replies in reviews, check out some `Standard replies
+  for reviewing
+  <https://scikit-learn.org/stable/developers/tips.html#saved-replies>`_.
 
 Reviewer Checklist
 ==================
 
-- Be patient and `kind <https://youtu.be/tzFWz5fiVKU?t=49m30s>`_ with
-  contributors.
-- When merging contributions, a committer is responsible for ensuring that those
-  meet the requirements outlined in the :ref:`Development process guidelines
-  <guidelines>` for NumPy. Also, check that new features and backwards
-  compatibility breaks were discussed on the `numpy-discussion mailing list
-  <https://mail.python.org/mailman/listinfo/numpy-discussion>`_.
 - Is the intended behavior clear under all conditions (e.g. unexpected inputs
   like empty arrays or nan/inf values)?
```

Comment:
This is too skimpy. That's ironic for an item that's urging thoroughness. We should give a fuller checklist of cases likely to be overlooked and suggestions on how to think of them generally. The list might be elsewhere and linked to here. 

We should also mention (possibly in a separate bullet) the `hypothesis` tool, for anyone who knows (or may want to learn) how to use it.

Reply:
Empty arrays and inf/nan are the main special cases. Nothing else immediately comes to mind - do you have any suggestions?

Reply:
Here are some more thoughts, some may be impractical/not appropriate
- axis or shape arguments should be int and tuples
- scalars and 0-d arrays
- there are techniques to use all the dtypes, which means that new ones (will we ever do another datetime?) will be added to the tests automatically when they are added to NumPy
- tests should hit the error cases in the code

Reply:
object arrays are huge special cases (e.g. array containing fractions/decimals).

Reply:
> object arrays are huge special cases (e.g. array containing fractions/decimals).

Yes, but do we really want to require tests with things like object arrays for decimals for all functionality? That seems like an unhealthy idea.

Reply:
+1 to all dtypes and axis int/tuple. the rest seems like overdoing it for a general review recommendation

Reply:
@rgommers I thought this was a "checklist" for what reviewers may want to look out for, not a strict requirement?

Reply:
Yes, but inevitably newer reviewers are going to ask for that stuff if it's on this list, and for something obscure like object array of decimals that's almost never desired. So making the list overly long isn't ideal imho.

Reply:
Can one of you explain the technique for testing all dtypes? I'm not familiar with it and don't really know how to explain. I'd say something like:

> Instead of listing specific dtypes when testing, *use technique*. This means that when new dtypes are added to NumPy, the tests will automatically include them.

Reply:
There's no standard way of grabbing a set of dtypes. Tests normally use `pytest.mark.parametrize`. For example:
```
int_types = [np.byte, np.short, np.intc, np.int_, np.longlong]
uint_types = [np.ubyte, np.ushort, np.uintc, np.uint, np.ulonglong]
float_types = [np.half, np.single, np.double, np.longdouble]
cfloat_types = [np.csingle, np.cdouble, np.clongdouble]


class TestArrayFromScalar:

    @pytest.mark.parametrize('t1', int_types + uint_types)
    @pytest.mark.parametrize('t2', int_types + uint_types + [None])
    def test_integers(self, t1, t2):
        return self._do_test(t1, t2)

    @pytest.mark.parametrize('t1', float_types)
    @pytest.mark.parametrize('t2', float_types + [None])
    def test_reals(self, t1, t2):
        return self._do_test(t1, t2)
```

The dtypes is something that requires some thought. Sometimes it uses `float32, float64`, or `"f4", "f8", ">f8",`. A search for `mark.parametrize` will show you a lot of ways of doing this.

Reply:
Actually, this is not the place to explain even `parametrize`. Maybe just add: _"are unusual dtypes tested if a function supports those?"_